backend/director/llm/xai.py

```
@@ -0,0 +1,181 @@
+class XAI(BaseLLM):
+    def __init__(self, config: XAIConfig = None):
+        """
+        :param config: XAI Config
+        """
+        if config is None:
+            config = XAIConfig()
+        super().__init__(config=config)
+        try:
+            import openai
+        except ImportError:
+            raise ImportError("Please install OpenAI python library.")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)